dev/com.ibm.tx.jta/src/com/ibm/tx/jta/impl/TransactionImpl.java

```diff
@@ -271,18 +273,34 @@ public void prepare() {
         }
     };
 
-    private static NewTransactionCheckpointHook getNewTxHook() {
+    private static NewTransactionCheckpointHook getNewTxCheckpointHook() {
```

Comment:
I think this is mis-named.  Calling it "New" implies you are returning a new object each call.  I suggest removing "New" from the method and class name.

Reply:
Renamed the hook class and encapsulated all related state and behaviors.